src/plugins/intel_gpu/src/graph/permute.cpp

```diff
@@ -143,6 +143,12 @@ void permute_inst::update_output_memory() {
 
     GPU_DEBUG_TRACE_DETAIL << id() << " : update_output_memory with mem of input " << get_node().get_dependency(0).id()
                            << " : " << input_memory_ptr()->buffer_ptr() << std::endl;
+    // Can_be_optimized nodes are allocating from memory_pool too. In this case,
+    // we need release the legacy output memory from memory pool explicitly.
+    if (static_cast<bool>(_outputs[0]) &&
+        _node->get_program().get_config().get_property(ov::intel_gpu::enable_memory_pool)) {
+        _network.get_memory_pool().release_memory(_outputs[0].get(), _node->get_unique_id(), _node->id(), _network.get_id());
+    }
```

Comment:
TODO:
We should also add this _node into the memory pool user list of new output memory to avoid the memory is dereferenced and freed. Root caused?

Reply:
I agree, could you apply them to other update_output_memory functions too?

Reply:
done